### PR TITLE
Add cycle lane styles

### PR DESCRIPTION
**Pull request by westnordost**
See https://github.com/streetcomplete/StreetComplete/issues/4502

Work in progress.

### CODE REVIEW

File: data/hasAdvisoryCycleLane.yml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+# see https://reconstruccionaccidentestrafico.com/vias-ciclistas-definicion-legal-normativa-y-prioridad-de-paso-respecto-a-otros-vehiculos-y-a-peatones/
+ES: false
+
+FI: false
+
+FR: true # https://duckduckgo.com/?q=chaucidou&t=ffab&iax=images&ia=images
+
+# https://road.cc/content/news/advisory-bike-lines-increase-risk-cyclist-casualties-279553
+GB: true # https://www.google.de/maps/@51.4957567,-0.12058,3a,75y,344.59h,64.38t/data=!3m6!1e1!3m4!1sJTZbilG681QyqAGUN7XfFw!2e0!7i16384!8i8192
+
```
Comment (mnalis) [inline data/hasAdvisoryCycleLane.yml:50]:
```suggestion

# Članak 46.(1), Članak 57.(6)
# https://narodne-novine.nn.hr/clanci/sluzbeni/2008_06_67_2224.html
HR: false
```

In Croatia on the roads we don't have `cycleway=lane`+`cycleway:lane=advisory`, only **mandatory bicycle lanes** _or_ **sharrows** (`cycleway=shared_lane`+`cycleway:lane=pictogram`). The law translates as:

> (1) The driver of a motor vehicle must use only the road, move in the middle of the marked road
traffic lane, i.e. the lane intended for the traffic of the vehicle category to which the vehicle belongs, except in the case danger to life, health and property.
> (6) A driver who, when turning, crosses a bicycle path or a lane that extends along the roadway he is moving on, is obliged to let bicycles moving along the bicycle path or lane in the same or opposite direction.

File: data/exclusiveCycleLaneStyle.yml [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+# https://finlex.fi/fi/laki/alkup/2018/20180729
+FI: white dashes
+
+FR: white dashes # https://www.google.de/maps/@48.852818,2.3674504,3a,75y,33.46h,64.09t/data=!3m6!1e1!3m4!1sait3Kj3mhuaT0PGthUYGQw!2e0!7i16384!8i8192
+
+GB: white # https://www.google.de/maps/@51.4965114,-0.1255504,3a,75y,346.83h,84.96t/data=!3m6!1e1!3m4!1sty8Zl-R_msKxFjnB702GBw!2e0!7i16384!8i8192
+
+# Zakon o sigurnosti prometa na cestama, Članak 18)
+# https://www.zakon.hr/z/78/Zakon-o-sigurnosti-prometa-na-cestama
+HR: yellow # https://www.google.de/maps/@45.8061213,15.9722997,3a,75y,118.67h,62.19t/data=!3m6!1e1!3m4!1sfJFX3JAwYpYKkjfujlN8jA!2e0!7i13312!8i6656
```
Comment (mnalis) [inline data/exclusiveCycleLaneStyle.yml:72]:
```suggestion
# Zakon o sigurnosti prometa na cestama, Članak  2. (1) 18), Članak 112. (1)
# https://narodne-novine.nn.hr/clanci/sluzbeni/2008_06_67_2224.html
# https://narodne-novine.nn.hr/clanci/sluzbeni/2022_07_85_1288.html
# Pravilnik o biciklističkoj infrastrukturi, Članak 40., 48.
# https://narodne-novine.nn.hr/clanci/sluzbeni/2016_03_28_803.html
HR: yellow # https://www.google.de/maps/@45.8090029,15.971595,3a,75y,243.6h,51.8t/data=!3m6!1e1!3m4!1skNmcpBP1iJdD5J_2ZGqEhQ!2e0!7i13312!8i6656
```

Um, I think current file is wrong for `HR`:
- `Članak 18.` (as quoted) seems unrelated, as it talks about semaphores and other lightning, not about cycle lanes? Maybe you though about `Članak  2. (1) 18)` which defines what bicycle lane is, or `Članak 112.(1)` which defines that cyclists must use it, but those only define **_existence_** of exclusive cycle lanes, not their style.

- Firstly, `www.zakon.hr` website is private company (whose business model is combining amendments to a law to a single text, and selling law precedent case decisions) and not official source (despite nice domain name they've grabbed on fist-come-first-served basis!) related to that, law texts on that site are mutable (e.g. what was `Članak 18.` yesterday, might be `Članak 19.` today, or be removed etc.). So best to use official and unmutable source, which is https://narodne-novine.nn.hr/clanci/sluzbeni/xxxx (the disadvantage there however is that sometimes one has to quote several URLs if some law has changed significantly)

- **_style_** of the cycling infrastructure is defined  by sub-law regulation in `.pdf` found at: https://narodne-novine.nn.hr/clanci/sluzbeni/2016_03_28_803.html, e.g. Članak 40., 48. However it only seems to define that the line must be full (not dashed), not it's color (it does say in what cases there must also be red surface). 
 Maybe I'm missing it, or it is defined in some other regulation :man_shrugging: 

  The actual practice on the ground itself is also various (but it does not overly surprise me, as even quite clear regulations about cycling infrastructure are being ignored all the time by the construction contractors without repercussions), but most recognizable would be `red surface+full yellow line` or just `full yellow line` for "cycle lane" on the road ("biciklistička traka" in Croatian):
  - red surface, full yellow line: https://www.google.de/maps/@45.8090029,15.971595,3a,75y,243.6h,51.8t/data=!3m6!1e1!3m4!1skNmcpBP1iJdD5J_2ZGqEhQ!2e0!7i13312!8i6656
  - just a full yellow line: https://www.google.de/maps/@45.8081707,15.972222,3a,75y,17.33h,62.09t/data=!3m6!1e1!3m4!1sTIjQdxBV4OUwZf1Gtw08XQ!2e0!7i13312!8i6656
  or https://www.mapillary.com/app/?lat=45.719591&lng=16.058816&z=17&pKey=1740656346112283&focus=photo
  or https://www.mapillary.com/app/?lat=45.55183606&lng=18.69876068&z=17&pKey=1218993981851809&focus=photo
  - just a full white line: https://www.mapillary.com/app/?lat=45.777645483610996&lng=15.916126705556053&z=17.656886226982298&pKey=949690755844107&focus=photo&x=0.530157050275831&y=0.5816257008

File: data/hasAdvisoryCycleLane.yml (hunk 1 above)
Comment (westnordost) [reply data/hasAdvisoryCycleLane.yml:50]: thank you!

File: data/exclusiveCycleLaneStyle.yml (hunk 2 above)
Comment (westnordost) [reply data/exclusiveCycleLaneStyle.yml:72]: Yeah, that's why I wrote "yellow" and not "yellow on red background", because it is not always the case.

**Pull request merged**